glTFLoader/Interface.cs

```diff
@@ -71,55 +78,155 @@ private static string ParseBinary(Stream stream)
         {
             using (BinaryReader binaryReader = new BinaryReader(stream))
             {
-                uint magic = binaryReader.ReadUInt32();
-                if (magic != GLTF)
-                {
-                    throw new InvalidDataException($"Unexpected magic number: {magic}");
-                }
+                ReadBinaryHeader(binaryReader);
+
+                var data =  ReadBinaryJsonChunk(binaryReader);
 
-                uint version = binaryReader.ReadUInt32();
-                if (version != 2)
-                {
-                    throw new NotImplementedException($"Unknown version number: {version}");
-                }
-                
-                uint length = binaryReader.ReadUInt32();
-                long fileLength = stream.Length;
-                if (length != fileLength)
-                {
-                    throw new InvalidDataException($"The specified length of the file ({length}) is not equal to the actual length of the file ({fileLength}).");
-                }
+                return Encoding.UTF8.GetString(data);
+            }
+        }
+
+        private static byte[] ReadBinaryJsonChunk(BinaryReader binaryReader)
+        {
+            uint chunkLength = binaryReader.ReadUInt32();
+            if ((chunkLength & 3) != 0)
+            {
+                throw new NotImplementedException($"The first chunk must be padded to 4 bytes: {chunkLength}");
+            }
+
+            uint chunkFormat = binaryReader.ReadUInt32();
+            if (chunkFormat != JSON)
+            {
+                throw new NotImplementedException($"The first chunk must be format 'JSON': {chunkFormat}");
+            }
+
+            return binaryReader.ReadBytes((int)chunkLength);
+        }
+
+        public static Byte[] LoadBinaryBuffer(string filePath)
+        {
+            using (Stream stream = File.OpenRead(filePath))
+            {
+                return LoadBinaryBuffer(stream);
+            }
+        }
+
+        public static Byte[] LoadBinaryBuffer(Stream stream)
+        {
+            using (BinaryReader binaryReader = new BinaryReader(stream))
+            {
+                ReadBinaryHeader(binaryReader);
 
+                // skip JSON chunk
+                ReadBinaryJsonChunk(binaryReader);
+
                 uint chunkLength = binaryReader.ReadUInt32();
-                uint chunkFormat = binaryReader.ReadUInt32();
-                if (chunkFormat != JSON)
-                {
-                    throw new NotImplementedException($"The first chunk must be format 'JSON': {chunkFormat}");
+                if ((chunkLength & 3) != 0)
+                {
+                    throw new NotImplementedException($"The second chunk must be padded to 4 bytes: {chunkLength}");

```

Comment:
NotImplementedException seems weird for this. Perhaps use the InvalidDataException you are using elsewhere?